src/org/sipdroid/sipua/RegisterAgent.java

```diff
@@ -155,6 +158,8 @@ public RegisterAgent(SipProvider sip_provider, String target_url,
 		this.pub = pub;
 		
 		this.mwi = mwi;
+
+		pas = new PublishAgent(sip_provider, user_profile,user_profile.username, user_profile.realm, user_profile.passwd, context);

```

Comment:
I noticed we pass many things from the user profile, and the user profile itself. The PublishAgent could take what it needs from user_profile instead of passing 3 extra things in the constructor.